web/src/main/frontend/src/components/Points.js

```
@@ -11,32 +11,46 @@ class Points extends React.Component {
 
   createInputField(fixed, correct, valuated) {
     const className = fixed && !correct ? 'no-spinner wrong' : 'no-spinner';
-    const value = fixed ? this.props.points : null;
+    const value = fixed
+      ? correct
+        ? `${this.props.points}/${this.props.points}`
+        : `0/${this.props.points}`
+      : undefined;
     const readOnly = fixed || null;
     const onWheel = valuated && this.turnOffScrollWheel;
     const onChange = valuated && this.onChange;
-    const placeholder = valuated ? this.props.points : null;
+    const onBlur = valuated && this.onBlur;
+    const onFocus = valuated && this.onFocus;
+    const placeholder = `0/${this.props.points}`;
 
     return (
       <input
         className={className}
-        value={value || placeholder}
+        value={value}
         readOnly={readOnly}
         onWheel={onWheel}
         onChange={onChange}
         placeholder={placeholder}
+        onBlur={onBlur}
+        onFocus={onFocus}
       />
     );
   }
 
   turnOffScrollWheel = () => e => e.preventDefault();
 
+  onFocus = e => {
+    if (e.target.value && e.target.value.includes('/')) {
+      e.target.value = parseInt(e.target.value.split('/')[0], 10);
```

We shouldn't modify `value` directly.

---

How should i modify the value of input? One of the use cases is that i want to set it to 0 if user tries tou enter less than 0 or I want to fall back to placeholder if input is cleared

---

Make it a controlled input imo - https://reactjs.org/docs/forms.html#controlled-components